astropy/cosmology/units.py

```diff
@@ -98,8 +97,15 @@ def redshift_distance(cosmology=None, kind="comoving", **atzkw):
     >>> from astropy.cosmology import WMAP9
 
     >>> z = 1100 * cu.redshift
-    >>> z.to(u.Mpc, cu.redshift_distance(WMAP9, kind="comoving"))  # doctest: +FLOAT_CMP
+    >>> d = z.to(u.Mpc, cu.redshift_distance(WMAP9, kind="comoving"))
+    >>> d  # doctest: +FLOAT_CMP
     <Quantity 14004.03157418 Mpc>
+
+    To go from distance to redshift it is necessary to adjust the maximum redshift of
+    :func:`~astropy.cosmology.z_at_value`.
```

Comment:
I would not mention `z_at_value()` by name here because it is not explicitly called in the example code.
```suggestion
    To convert a very large distance to a redshift it might be necessary to specify
    a large enough ``zmax`` value.
```

Reply:
Done!

Reply:
The updated phrasing is indeed better.